File: src/main/java/net/volcano/jdacommands/model/command/annotations/Help.java
package net.volcano.jdacommands.model.command.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface Help {
	
	String usage() default "GENERATE";
	
	String description();
	
	String category() default "DEFAULT";
	
	String[] examples() default {};
	
	String details() default "";
	
	Permission[] permissions() default {};
	
}

File: src/main/java/net/volcano/jdacommands/model/command/annotations/Help.kt
package net.volcano.jdacommands.model.command.annotations

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
annotation class Help(

	val usage: String = "GENERATE",

	val description: String,

	val category: String = "DEFAULT",

	val examples: Array<String> = [],

	val details: String = "",

	val permissions: Array<Permission> = []

)